### PR TITLE
API - Requester can set his task as finalized

### PR DESCRIPTION
**Pull request by janko101**
# [PT Story](https://www.pivotaltracker.com/story/show/172402314)
```
As a Requester
When I receive the products from a helper
I would like to be able to mark the task as finalized.
```
## Changes proposed in this pull request:
* Adds activity: "finalized" to enum status in a Task Model
* Restricts that only the Requester can mark a task as finalized in a Task Model
* Adds the update of activity status to finalized in a Tasks Controller.
*Happy and Sad Paths have been tested for the feature.

### CODE REVIEW

File: spec/requests/api/v1/requester/finalized_task_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    describe 'Another user tries to finalize the task' do
+      let!(:another_user) { create(:user) }
+      let!(:another_user_credentials) { another_user.create_new_auth_token }
+      let!(:another_user_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(another_user_credentials) }  
+
+      before do
+        put "/api/v1/tasks/#{task.id}",
+            params: { activity: 'finalized' },
```
Comment (blake-futchi) [inline spec/requests/api/v1/requester/finalized_task_spec.rb]: spacing

**Review by blake-futchi — COMMENTED** [review]
not many comments here. Looks good

File: app/controllers/api/v1/tasks_controller.rb [hunk 2]
```diff
@@ -39,6 +39,15 @@ def update
       else
         render_error_message(@task)
       end
+
+    when 'finalized'
+      if @task.is_finalizable?(current_user)
+        @task.update(status: params[:activity], user: current_user)
+        render json: { message: 'You finalized the delivery' }
```
Comment (pierre-1) [inline app/controllers/api/v1/tasks_controller.rb]: Perhaps: 'Your delivery is complete and you made someone happy :) Thanks for being a Helping Hander!"

Comment (blake-futchi) [reply app/controllers/api/v1/tasks_controller.rb]: finalized comes from the requester right?

Comment (blake-futchi) [reply app/controllers/api/v1/tasks_controller.rb]: So it should probably be: We are happy that you received your order. Please be in touch if you have any further questions.

**Comment by pierre-1** [comment]
Looking good, also glad to see that the sad paths were reusable!

File: app/controllers/api/v1/tasks_controller.rb (hunk 2 above)
Comment (janko101) [reply app/controllers/api/v1/tasks_controller.rb]: We gonna change all of those messages in one PR

**Pull request merged**